features/support/api_helpers.rb

```diff
@@ -95,7 +95,7 @@ def ensure_no_framework_agreements_exist(framework_slug)
 
 def update_framework_agreement_status(framework_slug, supplier_id, status)
   response = call_api(:post, "/suppliers/#{supplier_id}/frameworks/#{framework_slug}", payload: {
-    frameworkInterest: {agreementReturned: status},
+    frameworkInterest: {onFramework: status},
```

Comment:
Yep, this needs to be renamed  as something like "update_supplier_on_framework".

Agreement status can no longer be set using this API endpoint (as you've discovered).